lms/djangoapps/verify_student/tests/test_views.py

```diff
@@ -1885,3 +1980,250 @@ def _get_url(self, course_key, checkpoint):
+        self.assertIn("https://{}{}".format(
+            microsite.get_value('SITE_NAME', 'localhost'), self.re_verification_link), body
+        )
+
+    def test_denied_email_message_with_close_verification_dates(self):
+
+        return_value = datetime(2016, 1, 1, tzinfo=timezone.utc)
+        with patch.object(timezone, 'now', return_value=return_value):
+            subject, body = _compose_message_reverification_email(
+                self.course.id, self.user.id, "midterm", self.attempt, "denied", True
+            )
+
+            self.assertIn(
+                "Your verification for course {course_name} and assessment {assessment} has failed.".format(
+                course_name=self.course.display_name_with_default,
+                assessment=self.assessment
+                ),
+                body
+            )
+
+            self.assertIn("Assessment date has passed and retake not allowed", body)
+
+
+class TestEmailMessageWithDefaultICRVBlock(ModuleStoreTestCase):
+    """
+    Test for In-course Re-verification
+    """
+
+    def build_course(self):
+        """
+        Build up a course tree with a Reverificaiton xBlock.
+        """
+        # pylint: disable=attribute-defined-outside-init
+
+        self.course_key = SlashSeparatedCourseKey("Robot", "999", "Test_Course")
+        self.course = CourseFactory.create(org='Robot', number='999', display_name='Test Course')
+
+        # Create the course modes
+        for mode in ('audit', 'honor', 'verified'):
+            min_price = 0 if mode in ["honor", "audit"] else 1
+            CourseModeFactory(mode_slug=mode, course_id=self.course_key, min_price=min_price)
+
+        # Create the 'edx-reverification-block' in course tree
+        section = ItemFactory.create(parent=self.course, category='chapter', display_name='Test Section')
+        subsection = ItemFactory.create(parent=section, category='sequential', display_name='Test Subsection')
+        vertical = ItemFactory.create(parent=subsection, category='vertical', display_name='Test Unit')
+
+        self.reverification = ItemFactory.create(
+            parent=vertical,
+            category='edx-reverification-block',
+            display_name='Test Verification Block'
+        )
+
+        self.section_location = section.location
+        self.subsection_location = subsection.location
+        self.vertical_location = vertical.location
+        self.reverification_location = self.reverification.location
+        self.assessment = "midterm"
+
+        self.re_verification_link = reverse(
+            'verify_student_incourse_reverify',
+            args=(
+                unicode(self.course_key),
+                unicode(self.assessment),
+                unicode(self.reverification_location)
+            )
+        )
+
+    def setUp(self):
+        super(TestEmailMessageWithDefaultICRVBlock, self).setUp()
+
+        self.build_course()
+        self.check_point1 = VerificationCheckpoint.objects.create(
```

Comment:
Since we are only using one checkpoint so change `self.check_point1` to just `check_point`.